File: src/Creator/CreateLecturer.java
package Creator;

import PersonObject.*;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;

public class CreateLecturer {
    private ArrayList<Lecturer> lecturers = new ArrayList<>();
    private String fileName;

    public CreateLecturer(String fileName) {
        this.fileName = fileName;
        createLecturers();
    }
    public void createLecturers() {
        try {
            String content = new String(Files.readAllBytes(Path.of(fileName)));
            JSONObject jsonObject = new JSONObject(content);
            JSONArray lecturerJSON = jsonObject.getJSONArray("lecturers");
            for (int i = 0; i < lecturerJSON.length(); i++) {
                String name = lecturerJSON.getJSONObject(i).getString("name");
                String surname = lecturerJSON.getJSONObject(i).getString("surname");
                String lecturerId = lecturerJSON.getJSONObject(i).getString("lecturerId");
                String password = lecturerJSON.getJSONObject(i).getString("password");
                lecturers.add(new Lecturer(name, surname, new Id(lecturerId), new Password(password)));
            }
        }
        catch (JSONException | IOException ignored){
            System.err.println("An error occurred in the lecturers JSON file. Please ensure that the file is created in the correct format and fix any errors.");
            System.exit(0);
        }
    }

    public ArrayList<Lecturer> getLecturers() {
        return lecturers;
    }
}

File: python_development/Creator/CreateLecturer.py
import json

from python_development.Lecturer import Lecturer
from python_development.Id import Id
from python_development.Password import Password

class CreateLecturer:
    def __init__(self, fileName):
        self.lecturers = []
        self.fileName = fileName
        self.create_lecturers()

    def create_lecturers(self):
        try:
            with open(self.fileName, 'r') as file:
                data = json.load(file)
                lecturers_data = data.get('lecturers', [])
                for lecturer_info in lecturers_data:
                    name = lecturer_info['name']
                    surname = lecturer_info['surname']
                    lecturer_id = lecturer_info['lecturerId']
                    password = lecturer_info['password']
                    lecturer = Lecturer(name, surname, Id(lecturer_id), Password(password))
                    self.lecturers.append(lecturer)
        except (json.JSONDecodeError, FileNotFoundError) as e:
            print(f"An error occurred in the lecturers JSON file: {e}. Please ensure the file is in the correct format.")

    def getLecturers(self):
        return self.lecturers
